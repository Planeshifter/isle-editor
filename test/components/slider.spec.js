// MODULES //

import React from 'react';
import renderer from 'react-test-renderer';
import Slider from 'components/slider';


// TESTS //

it( 'renders a slider for child elements', () => {
	const slider = renderer.create(
		<Slider>
			<div>First Page</div>
			<div>Second Page</div>
			<div>Third Page</div>
		</Slider>
	).toJSON();
	expect( slider ).toMatchSnapshot();
});

it( 'renders a slider for child elements (dragging disabled)', () => {
	const slider = renderer.create(
		<Slider draggable={false} >
			<div>First Page</div>
			<div>Second Page</div>
			<div>Third Page</div>
		</Slider>
	).toJSON();
	expect( slider ).toMatchSnapshot();
});

it( 'renders a slider for child elements (with title)', () => {
	const slider = renderer.create(
		<Slider title="Questions" >
			<div>First Page</div>
			<div>Second Page</div>
			<div>Third Page</div>
		</Slider>
	).toJSON();
	expect( slider ).toMatchSnapshot();
});

it( 'renders a slider for child elements (without dots for navigation)', () => {
	const slider = renderer.create(
		<Slider dots={false} >
			<div>First Page</div>
			<div>Second Page</div>
			<div>Third Page</div>
		</Slider>
	).toJSON();
	expect( slider ).toMatchSnapshot();
});
