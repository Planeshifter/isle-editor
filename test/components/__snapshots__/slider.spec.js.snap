// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a slider for child elements (dragging disabled) 1`] = `
<div
  className="card"
  size="large"
  style={
    Object {
      "marginBottom": "20px",
      "marginLeft": "auto",
      "marginRight": "auto",
      "marginTop": "20px",
      "maxWidth": "1200px",
      "width": "100%",
    }
  }
>
  <div
    className="card-body"
    style={
      Object {
        "paddingBottom": "40px",
      }
    }
  >
    <div
      beforeChange={[Function]}
      className="centered"
      dots={true}
      draggable={false}
      fade={false}
      goto={0}
      infinite={false}
      interval={null}
      nextArrow={
        <NextArrow
          className=""
          onClick={[Function]}
        />
      }
      onClick={[Function]}
      prevArrow={
        <PreviousArrow
          className=""
          onClick={[Function]}
        />
      }
      slidesToScroll={1}
      slidesToShow={1}
      speed={1000}
      style={Object {}}
      swipe={false}
      t={[Function]}
      title=""
    >
      0
    </div>
  </div>
</div>
`;

exports[`renders a slider for child elements (with title) 1`] = `
<div
  className="card"
  size="large"
  style={
    Object {
      "marginBottom": "20px",
      "marginLeft": "auto",
      "marginRight": "auto",
      "marginTop": "20px",
      "maxWidth": "1200px",
      "width": "100%",
    }
  }
>
  <h3
    className="card-header"
  >
    Questions
    <span
      style={
        Object {
          "float": "right",
        }
      }
    >
      1
       / 
      3
    </span>
  </h3>
  <div
    className="card-body"
    style={
      Object {
        "paddingBottom": "40px",
      }
    }
  >
    <div
      beforeChange={[Function]}
      className="centered"
      dots={true}
      draggable={true}
      fade={false}
      goto={0}
      infinite={false}
      interval={null}
      nextArrow={
        <NextArrow
          className=""
          onClick={[Function]}
        />
      }
      onClick={[Function]}
      prevArrow={
        <PreviousArrow
          className=""
          onClick={[Function]}
        />
      }
      slidesToScroll={1}
      slidesToShow={1}
      speed={1000}
      style={Object {}}
      swipe={false}
      t={[Function]}
      title="Questions"
    >
      0
    </div>
  </div>
</div>
`;

exports[`renders a slider for child elements (without dots for navigation) 1`] = `
<div
  className="card"
  size="large"
  style={
    Object {
      "marginBottom": "20px",
      "marginLeft": "auto",
      "marginRight": "auto",
      "marginTop": "20px",
      "maxWidth": "1200px",
      "width": "100%",
    }
  }
>
  <div
    className="card-body"
    style={
      Object {
        "paddingBottom": "40px",
      }
    }
  >
    <div
      beforeChange={[Function]}
      className="centered"
      dots={false}
      draggable={true}
      fade={false}
      goto={0}
      infinite={false}
      interval={null}
      nextArrow={
        <NextArrow
          className=""
          onClick={[Function]}
        />
      }
      onClick={[Function]}
      prevArrow={
        <PreviousArrow
          className=""
          onClick={[Function]}
        />
      }
      slidesToScroll={1}
      slidesToShow={1}
      speed={1000}
      style={Object {}}
      swipe={false}
      t={[Function]}
      title=""
    >
      0
    </div>
  </div>
</div>
`;

exports[`renders a slider for child elements 1`] = `
<div
  className="card"
  size="large"
  style={
    Object {
      "marginBottom": "20px",
      "marginLeft": "auto",
      "marginRight": "auto",
      "marginTop": "20px",
      "maxWidth": "1200px",
      "width": "100%",
    }
  }
>
  <div
    className="card-body"
    style={
      Object {
        "paddingBottom": "40px",
      }
    }
  >
    <div
      beforeChange={[Function]}
      className="centered"
      dots={true}
      draggable={true}
      fade={false}
      goto={0}
      infinite={false}
      interval={null}
      nextArrow={
        <NextArrow
          className=""
          onClick={[Function]}
        />
      }
      onClick={[Function]}
      prevArrow={
        <PreviousArrow
          className=""
          onClick={[Function]}
        />
      }
      slidesToScroll={1}
      slidesToShow={1}
      speed={1000}
      style={Object {}}
      swipe={false}
      t={[Function]}
      title=""
    >
      0
    </div>
  </div>
</div>
`;
